Weekly update of JEDI hashes (20250819)

RussTreadon-NOAA opened the pull request:
# Description

This PR includes two sets of changes:
1. weekly update of JEDI hashes
2. updates that allow jcb to be built and installed from compute nodes

# Companion PRs

g-w PR [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951) depends upon closure of this PR.   Once this PR is merged into GDASApp `develop`, the `sorc/gdas.cd` hash in g-w R [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951) will be updated to the new GDASApp `develop` hash.

# Issues

Resolves #1849
Resolves #1852
Resolves #1862
Partially addresses #1851 - this PR does not address the external `fetch` dependency added by the updated ioda-converters hash.

# Automated CI tests to run in Global Workflow
- [ ] atm_jjob <!-- JEDI atm single cycle DA !-->
- [ ] C96C48_ufs_hybatmDA <!-- JEDI atm cycled DA !-->
- [ ] C96C48_hybatmsnowDA <!-- JEDI snow cycled DA !-->
- [ ] C96_gcafs_cycled <!-- JEDI aerosol cycled DA !-->
- [ ] C48mx500_3DVarAOWCDA <!-- JEDI low-res marine 3DVar cycled DA  !-->
- [ ] C48mx500_hybAOWCDA <!-- JEDI marine hybrid envar cycled DA !-->
- [ ] C96C48_hybatmDA <!-- GSI atm cycled DA !-->

Note:  The updated `sorc/soca` hash in this PR requires an updated `ufs_model` hash in g-w.  The updated `ufs_model` hash is in g-w PR [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951).

As such, we cannot run automated GDASApp CI on this PR.  Automated GDASApp g-w CI clones g-w `develop`.  g-w `develop` does not, at present, point at a `ufs_model` hash which works with the updated `sorc/soca` hash in `feature/stable-nightly`.

RussTreadon-NOAA committed:
Update to new stable build on 20250812

RussTreadon-NOAA committed:
compute node pip install patch

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA committed:
Update to new stable build on 20250814

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA committed:
Update to new stable build on 20250815

RussTreadon-NOAA commented:
Mark this PR as _draft_ until testing with g-w PR [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951) is complete.

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA commented:
**Hera ctests**

Install `feature/stable-nightly` at [f644802](https://github.com/NOAA-EMC/GDASApp/commit/f644802cc3276830a32fcc2e2da7d732bb79cdee) inside `JessicaMeixner-NOAA:ModelUpdateAfterSpack` at [14fab79](https://github.com/JessicaMeixner-NOAA/global-workflow/commit/14fab7940108a652f89f30a0ca1dee61ebdec343).  This g-w branch is described in g-w PR [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951).  

Run ctests with following results
```
97% tests passed, 5 tests failed out of 157

Label Time Summary:
gdas-utils    =   6.36 sec*proc (12 tests)
gdasapp       = 18666.71 sec*proc (116 tests)
script        = 18673.07 sec*proc (128 tests)

Total Test time (real) = 2695.09 sec

The following tests FAILED:
        2109 - test_gdasapp_C96_gcafs_cycled_gcdas_aeroanlgenb_202112201200 (Failed)
        2113 - test_gdasapp_C96_gcafs_cycled_gcdas_aeroanlgenb_202112201800 (Failed)
        2114 - test_gdasapp_C96_gcafs_cycled_gcdas_aeroanlinit_202112201800 (Failed)
        2115 - test_gdasapp_C96_gcafs_cycled_gcdas_aeroanlvar_202112201800 (Failed)
        2116 - test_gdasapp_C96_gcafs_cycled_gcdas_aeroanlfinal_202112201800 (Failed)
```

Examine `gcdas_aeroanlgenb` failure.  Job fails due to problem rendering an input template.  See GDASApp issue #1862 for details.

This PR can not move forward until the problem with `gcdas_aeroanlgenb` is resolved.

RussTreadon-NOAA commented:
A solution for the `gcdas_aeroanlgenb` failure has been identified and tested (see GDASApp issue #1862 for details).  Changes are required to `config.aeroanl.j2` and `config.aeroanlgenb` in g-w `dev/parm/config/gcafs/`

RussTreadon-NOAA commented:
@JessicaMeixner-NOAA added the updated config files to g-w PR [#3591](https://github.com/NOAA-EMC/global-workflow/pull/3951#issuecomment-3197325045).  Given this, trigger Hera stable-nightly job using g-w PR [#3591](https://github.com/NOAA-EMC/global-workflow/pull/3951#issuecomment-3197325045).  GDASApp ctest tesults will be posted in this PR when available.

RussTreadon-NOAA commented:
**WCOSS GDASApp ctests**

Install `JessicaMeixner-NOAA:ModelUpdateAfterSpack` at [5eedaf5](https://github.com/JessicaMeixner-NOAA/global-workflow/commit/5eedaf5adeee2e089058ccc7c41f64d2507da562) on Cactus with `feature/stable-nightly` at f644802.  Run GDASApp ctests with following results
```
nohup: ignoring input
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/pr3951/sorc/gdas.cd/build
        Start 1943: test_gdasapp_jcb_installation
        Start 1967: test_gdasapp_C96C48_ufs_hybatmDA
        Start 1948: test_gdasapp_C96C48_hybatmDA
        Start 2006: test_gdasapp_C96C48_hybatmsnowDA
        Start 1991: test_gdasapp_C96_gcafs_cycled
        Start 2038: test_gdasapp_C48mx500_hybAOWCDA
        Start 2027: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2054: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1557: test_gdasapp_util_prepdata
        Start 1946: test_gdasapp_snow_apply_jediincr
        Start 2053: test_gdasapp_convert_gsi_satbias
        Start 1556: test_gdasapp_util_coding_norms
  1/136 Test #1943: test_gdasapp_jcb_installation ............................................***Failed    2.31 sec
        Start 1941: test_gdasapp_check_python_norms
  2/136 Test #1557: test_gdasapp_util_prepdata ...............................................   Passed    5.16 sec
        Start 1945: test_gdasapp_snow_imsproc
  3/136 Test #2054: test_gdasapp_setup_atm_jjob_cycled_exp ...................................   Passed    8.00 sec
...
136/136 Test #1990: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  267.08 sec

99% tests passed, 1 tests failed out of 136

Label Time Summary:
gdas-utils    =  21.01 sec*proc (10 tests)
gdasapp       = 24083.66 sec*proc (116 tests)
script        = 24104.67 sec*proc (126 tests)

Total Test time (real) = 3119.12 sec

The following tests FAILED:
        1943 - test_gdasapp_jcb_installation (Failed)
```

Rerun `test_gdasapp_jcb_installation` with `-VV` option.
```
1943: Testing jcb command...
1943: Traceback (most recent call last):
1943:   File "/lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/pr3951/sorc/gdas.cd/install/bin/jcb", line 5, in <module>
1943:     from jcb.driver import main
1943:   File "/lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/pr3951/sorc/gdas.cd/install/lib/python3.11/site-packages/jcb/driver.py", line 3, in <module>
1943:     import click
1943: ModuleNotFoundError: No module named 'click'
```

Module `py-click/8.1.7` is missing from the WCOSS2 `spack-stack/1.9.2` installation. This module is included in the spack-stack/1.9.2` installation on NOAA RDHPCS machines.

Open WCOSS2-requests issues [#46](https://github.com/NOAA-EMC/WCOSS2-requests/issues/46) to report this problem and [#47](
https://github.com/NOAA-EMC/WCOSS2-requests/issues/47) to request that `py-click/8.1.7` be added to WCOSS2 `spack-stack/1.9.2`.

RussTreadon-NOAA committed:
Update to new stable build on 20250818

shlyaeva commented:
I see soca tests are currently broken after https://github.com/JCSDA-internal/oops/pull/2968, it would be unsafe for marine application to update oops version until that's fixed.

RussTreadon-NOAA committed:
Update to new stable build on 20250819

RussTreadon-NOAA commented:
Thank you @shlyaeva for the heads up.

RussTreadon-NOAA committed:
add py-click to gaeac5 and gaeac6 modulefiles/GDAS, remove py-wheel from all moduefiles/GDAS

RussTreadon-NOAA commented:
**Multi-platform ctests**

Install `feature/stable-nightly` at 4fb0227 inside g-w `JessicaMeixner-NOAA:ModelUpdateAfterSpack` at [5eedaf5](https://github.com/JessicaMeixner-NOAA/global-workflow/commit/5eedaf5adeee2e089058ccc7c41f64d2507da562) on WCOSS2 (Cactus), Gaea C6, Hera, Hercules, Orion, and Ursa.  Run GDASApp ctests with the following results

**WCOSS2 (Cactus): all tests _Passed_**
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/pr3951/sorc/gdas.cd/build
        Start 1967: test_gdasapp_C96C48_ufs_hybatmDA
        Start 1948: test_gdasapp_C96C48_hybatmDA
        Start 1991: test_gdasapp_C96_gcafs_cycled
        Start 2006: test_gdasapp_C96C48_hybatmsnowDA
        Start 2027: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2038: test_gdasapp_C48mx500_hybAOWCDA
        Start 2054: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1557: test_gdasapp_util_prepdata
        Start 1556: test_gdasapp_util_coding_norms
        Start 1941: test_gdasapp_check_python_norms
        Start 1942: test_gdasapp_check_yaml_keys
        Start 1943: test_gdasapp_jcb_installation
  1/136 Test #1942: test_gdasapp_check_yaml_keys .............................................   Passed    1.52 sec
        Start 1944: test_gdasapp_snow_create_ens
  2/136 Test #1943: test_gdasapp_jcb_installation ............................................   Passed    3.02 sec

...

135/136 Test #1982: test_gdasapp_C96C48_ufs_hybatmDA_gdas_fcst_202402240000 ..................   Passed  424.22 sec
136/136 Test #1990: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  302.46 sec

100% tests passed, 0 tests failed out of 136

Label Time Summary:
gdas-utils    =  23.27 sec*proc (10 tests)
gdasapp       = 22442.43 sec*proc (116 tests)
script        = 22465.71 sec*proc (126 tests)

Total Test time (real) = 3203.22 sec
```
`test_gdasapp_jcb_installation` now passes because `py-click/8.1.7` has been included in the WCOSS2 spack-stack/1.9.2.  See WCOSS2-requests issue [#47](https://github.com/NOAA-EMC/WCOSS2-requests/issues/47) for details.

**Gaea C6:  all tests _Passed_**
```
Test project /gpfs/f6/ira-sti/scratch/Russ.Treadon/git/global-workflow/pr3951/sorc/gdas.cd/build
        Start 2070: test_gdasapp_C96C48_ufs_hybatmDA
        Start 2051: test_gdasapp_C96C48_hybatmDA
        Start 2094: test_gdasapp_C96_gcafs_cycled
        Start 2109: test_gdasapp_C96C48_hybatmsnowDA
        Start 2130: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2141: test_gdasapp_C48mx500_hybAOWCDA
        Start 2157: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1659: test_gdasapp_util_prepdata
        Start 1658: test_gdasapp_util_coding_norms
        Start 2044: test_gdasapp_check_python_norms
        Start 2045: test_gdasapp_check_yaml_keys
        Start 2046: test_gdasapp_jcb_installation
  1/136 Test #2045: test_gdasapp_check_yaml_keys .............................................   Passed    0.83 sec
        Start 2047: test_gdasapp_snow_create_ens
  2/136 Test #2046: test_gdasapp_jcb_installation ............................................   Passed    0.93 sec

...

135/136 Test #2129: test_gdasapp_C96C48_hybatmsnowDA_enkfgdas_fcst_202112201800 ..............   Passed  192.58 sec
136/136 Test #2093: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  168.50 sec

100% tests passed, 0 tests failed out of 136

Label Time Summary:
gdas-utils    =  27.13 sec*proc (10 tests)
gdasapp       = 13278.42 sec*proc (116 tests)
script        = 13305.56 sec*proc (126 tests)

Total Test time (real) = 1847.21 sec
```

**Hera:  all tests _Passed_**
```
Test project /scratch3/NCEPDEV/da/Russ.Treadon/CI/GDASApp/stable/20250819/global-workflow/sorc/gdas.cd/build
        Start 2081: test_gdasapp_C96C48_ufs_hybatmDA
        Start 2062: test_gdasapp_C96C48_hybatmDA
        Start 2105: test_gdasapp_C96_gcafs_cycled
        Start 2120: test_gdasapp_C96C48_hybatmsnowDA
        Start 2141: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2152: test_gdasapp_C48mx500_hybAOWCDA
        Start 2168: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1658: test_gdasapp_utils_incrhandler
        Start 1659: test_gdasapp_utils_incrhandler_8pes
        Start 1668: test_gdasapp_util_prepdata
        Start 1661: test_gdasapp_utils_hybridweights
        Start 1662: test_gdasapp_utils_setcorscales
  1/157 Test #2168: test_gdasapp_setup_atm_jjob_cycled_exp ...................................   Passed    1.28 sec
        Start 2173: test_gdasapp_atm_jjob_ens_init
  2/157 Test #1668: test_gdasapp_util_prepdata ...............................................   Passed    1.28 sec

...

156/157 Test #2104: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  245.44 sec
157/157 Test #2096: test_gdasapp_C96C48_ufs_hybatmDA_gdas_fcst_202402240000 ..................   Passed  434.65 sec

100% tests passed, 0 tests failed out of 157

Label Time Summary:
gdas-utils    =   6.30 sec*proc (12 tests)
gdasapp       = 18849.37 sec*proc (116 tests)
script        = 18855.67 sec*proc (128 tests)

Total Test time (real) = 2763.00 sec
```

**Hercules:  all tests _Passed_**
```
Test project /work/noaa/da/rtreadon/git/global-workflow/pr3951/sorc/gdas.cd/build
        Start 2072: test_gdasapp_C96C48_ufs_hybatmDA
        Start 2053: test_gdasapp_C96C48_hybatmDA
        Start 2111: test_gdasapp_C96C48_hybatmsnowDA
        Start 2132: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2143: test_gdasapp_C48mx500_hybAOWCDA
        Start 2159: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1659: test_gdasapp_util_prepdata
        Start 1658: test_gdasapp_util_coding_norms
        Start 1661: test_gdasapp_util_rtofstmp
        Start 1662: test_gdasapp_util_rtofssal
        Start 2046: test_gdasapp_check_python_norms
        Start 2047: test_gdasapp_check_yaml_keys
  1/133 Test #2047: test_gdasapp_check_yaml_keys .............................................   Passed    0.16 sec
        Start 2048: test_gdasapp_jcb_installation
  2/133 Test #2048: test_gdasapp_jcb_installation ............................................   Passed    0.40 sec

...

132/133 Test #2087: test_gdasapp_C96C48_ufs_hybatmDA_gdas_fcst_202402240000 ..................   Passed  446.73 sec
133/133 Test #2095: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  313.61 sec

100% tests passed, 0 tests failed out of 133

Label Time Summary:
gdas-utils    =  19.08 sec*proc (12 tests)
gdasapp       = 59446.92 sec*proc (101 tests)
script        = 59465.99 sec*proc (113 tests)

Total Test time (real) = 7985.62 sec
```

**Orion:  all tests _Passed_**
```
Test project /work2/noaa/da/rtreadon/git/global-workflow/pr3951/sorc/gdas.cd/build
        Start 2072: test_gdasapp_C96C48_ufs_hybatmDA
        Start 2053: test_gdasapp_C96C48_hybatmDA
        Start 2111: test_gdasapp_C96C48_hybatmsnowDA
        Start 2132: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2143: test_gdasapp_C48mx500_hybAOWCDA
        Start 2159: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1659: test_gdasapp_util_prepdata
        Start 1658: test_gdasapp_util_coding_norms
        Start 1661: test_gdasapp_util_rtofstmp
        Start 1662: test_gdasapp_util_rtofssal
        Start 2046: test_gdasapp_check_python_norms
        Start 2047: test_gdasapp_check_yaml_keys
  1/133 Test #2047: test_gdasapp_check_yaml_keys .............................................   Passed    0.23 sec
        Start 2048: test_gdasapp_jcb_installation
  2/133 Test #2048: test_gdasapp_jcb_installation ............................................   Passed    0.70 sec

...

132/133 Test #2095: test_gdasapp_C96C48_ufs_hybatmDA_enkfgdas_fcst_202402240000 ..............   Passed  285.76 sec
133/133 Test #2087: test_gdasapp_C96C48_ufs_hybatmDA_gdas_fcst_202402240000 ..................   Passed  471.64 sec

100% tests passed, 0 tests failed out of 133

Label Time Summary:
gdas-utils    =  13.31 sec*proc (12 tests)
gdasapp       = 22975.94 sec*proc (101 tests)
script        = 22989.25 sec*proc (113 tests)

Total Test time (real) = 4579.46 sec
```

**Ursa:  all tests _Passed_**
```
Test project /scratch3/NCEPDEV/da/Russ.Treadon/git/global-workflow/ursa/sorc/gdas.cd/build
        Start 2081: test_gdasapp_C96C48_ufs_hybatmDA
        Start 2062: test_gdasapp_C96C48_hybatmDA
        Start 2105: test_gdasapp_C96_gcafs_cycled
        Start 2120: test_gdasapp_C96C48_hybatmsnowDA
        Start 2141: test_gdasapp_C48mx500_3DVarAOWCDA
        Start 2152: test_gdasapp_C48mx500_hybAOWCDA
        Start 2168: test_gdasapp_setup_atm_jjob_cycled_exp
        Start 1658: test_gdasapp_utils_incrhandler
        Start 1659: test_gdasapp_utils_incrhandler_8pes
        Start 1668: test_gdasapp_util_prepdata
        Start 1661: test_gdasapp_utils_hybridweights
        Start 1662: test_gdasapp_utils_setcorscales
  1/157 Test #2168: test_gdasapp_setup_atm_jjob_cycled_exp ...................................   Passed    0.84 sec
        Start 2173: test_gdasapp_atm_jjob_ens_init
  2/157 Test #1668: test_gdasapp_util_prepdata ...............................................   Passed    0.93 sec

...

156/157 Test #2133: test_gdasapp_C96C48_hybatmsnowDA_gdas_fcst_202112201800 ..................   Passed  672.62 sec
157/157 Test #2096: test_gdasapp_C96C48_ufs_hybatmDA_gdas_fcst_202402240000 ..................   Passed  567.46 sec

100% tests passed, 0 tests failed out of 157

Label Time Summary:
gdas-utils    =   5.83 sec*proc (12 tests)
gdasapp       = 15850.57 sec*proc (116 tests)
script        = 15856.40 sec*proc (128 tests)

Total Test time (real) = 2434.04 sec
```
The following filers were modified in a local copy of `JessicaMeixner-NOAA:ModelUpdateAfterSpack` in order for all GDASApp tests to run on Ursa
```
        modified:   dev/parm/config/gfs/config.resources.URSA
        modified:   dev/workflow/hosts/ursa.yaml
        modified:   env/URSA.env
```
These changes will be documented in g-w issue [#3674](https://github.com/NOAA-EMC/global-workflow/issues/3674).

RussTreadon-NOAA commented:
GDASApp ctests have been run on multiple platforms with _Passed_ results for all tests run on each machine.  This PR is ready for review.

CoryMartin-NOAA reviewed (commented):
Any chance we can revert the iodaconv hash to allow it to compile on compute nodes? Is there anything recent that we need from a new iodaconv hash?

RussTreadon-NOAA commented:
@CoryMartin-NOAA

Yes, the ioda-converters hash can be reverted to ioda-converters at [48c1e3a](https://github.com/JCSDA-internal/ioda-converters/commit/48c1e3a2d6ba6ee41be22ad81f3ce67973f66922).  This is the hash prior to ioda-converters PR [#1672](https://github.com/JCSDA-internal/ioda-converters/pull/1672) being merged into ioda-converters `develop`.  This PR added the external dependency on [https://github.com/jbeder/yaml-cpp.git](https://github.com/jbeder/yaml-cpp.git)   

I'll need to rerun ctests on at least one machine to ensure no adverse consequences.

RussTreadon-NOAA commented:
@CoryMartin-NOAA , based on your [comments](https://github.com/NOAA-EMC/GDASApp/issues/1851#issuecomment-3201473408) in issue [#1851](https://github.com/NOAA-EMC/GDASApp/issues/1851), I'll go ahead an merge this PR into `develop`.  Doing so allows g-w PR [#3951](https://github.com/NOAA-EMC/global-workflow/pull/3951) to move forward.
